packages/calypso-e2e/src/lib/pages/published/published-post-page.ts

```diff
@@ -39,51 +30,52 @@ export class PublishedPostPage {
 		await Promise.all( [ this.page.waitForNavigation(), this.page.keyboard.press( 'Enter' ) ] );
 	}
 
-	/**
-	 * Returns the frame which holds the Like widget.
-	 *
-	 * @returns {Promise<Frame>} Frame holding the like widget on page.
-	 */
-	private async getLikeFrame(): Promise< Frame > {
-		// Obtain the ElementHandle for the widget containing the like/unlike button.
-		const elementHandle = await this.page.waitForSelector( selectors.likeWidget );
-		// Without the next line, headless viewports fail to locate the Like widget.
-		await elementHandle.scrollIntoViewIfNeeded();
-		// Obtain the Frame object from the elementHandleHandle. This represents the widget iframe.
-		const frame = ( await elementHandle.contentFrame() ) as Frame;
-		// Wait until the widget element is stable in the DOM.
-		await elementHandle.waitForElementState( 'stable' );
-
-		return frame;
-	}
-
 	/**
 	 * Clicks the Like button on the post.
 	 *
 	 * This method will also confirm that click action on the Like button
 	 * had the intended effect.
-	 *
-	 * @returns {Promise<void>} No return value.
 	 */
 	async likePost(): Promise< void > {
-		await this.page.evaluate( () => window.scrollTo( 0, document.body.scrollHeight ) );
-		const frame = await this.getLikeFrame();
-		await frame.click( selectors.likeButton );
-		await frame.waitForSelector( selectors.likedText, { state: 'visible' } );
+		const locator = this.page
+			.frameLocator( 'iframe[title="Like or Reblog"]' )
+			.getByRole( 'link', { name: 'Like' } );
+
+		// On AT sites Playwright is not able to scroll directly to the iframe
+		// containing the Like/Unlike button (similar to Post Comments).
+		await locator.evaluate( ( element ) => element.scrollIntoView() );
+
+		await locator.click();
+
+		// The button should now read "Liked".
+		await this.page
+			.frameLocator( 'iframe[title="Like or Reblog"]' )
+			.getByRole( 'link', { name: 'Liked' } )
+			.waitFor();
 	}
 
 	/**
 	 * Clicks the already-liked Like button on the post to unlike.
 	 *
 	 * This method will also confirm that click action on the Like button
 	 * had the intended effect.
-	 *
-	 * @returns {Promise<void>} No return value.
 	 */
 	async unlikePost(): Promise< void > {
-		const frame = await this.getLikeFrame();
-		await frame.click( selectors.unlikeButton );
-		await frame.waitForSelector( selectors.notLikedText, { state: 'visible' } );
+		const locator = this.page
+			.frameLocator( 'iframe[title="Like or Reblog"]' )
+			.getByRole( 'link', { name: 'Liked' } );
+
+		// On AT sites Playwright is not able to scroll directly to the iframe
+		// containing the Like/Unlike button (similar to Post Comments).
+		await locator.evaluate( ( element ) => element.scrollIntoView() );
+
+		await locator.click();
+
+		// The button should now read "Like".
+		await this.page
+			.frameLocator( 'iframe[title="Like or Reblog"]' )
+			.getByRole( 'link', { name: 'Like' } )
```

Comment:
Same as above, I'd apply the `exact` match here.

```suggestion
			.getByRole( 'link', { name: 'Like', exact: true } )
```

Reply:
Also fixed in eea8d6bcc30f4acb9b1e6def75ef20a599472aca.